ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/ip_access_list.py

```diff
@@ -0,0 +1,184 @@
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+from functools import cached_property
+from ipaddress import ip_network
+from typing import List
+
+from .utils import UtilsMixin
+
+
+class IPAccessListsMixin(UtilsMixin):
+    """
+    Mixin Class used to generate structured config for one key.
+    Class should only be used as Mixin to a AvdStructuredConfig class
+    """
+
+    sequence = 0
+
+    @cached_property
+    def ip_access_lists(self) -> list | None:
+        """
+        Return structured config for ip_access_lists
+
+        Used for the wan interfaces.
+        """
+
+        if self.shared_utils.wan_role is None:
+            return None
+
+        ip_access_lists = []
+        if acl_inbound_lists := self._get_access_lists_inbound():
+            ip_access_lists.extend(acl_inbound_lists)
+        if acl_outbound := self._get_access_list_outbound():
+            ip_access_lists.append(acl_outbound)
+
+        if ip_access_lists:
+            return ip_access_lists
+
+        return None
+
+    def _get_access_list_outbound(self) -> dict | None:
+        """
+        Returns the outbound access list.
+        Blocks the RFC1918 traffic.
+        """
+        if self.shared_utils.wan_interfaces:
+            outbound_access_list = {
+                "name": "WAN_TRANSPORT_OUT",
+                "entries": [
+                    {"sequence": self._get_sequence(True), "remark": " ### Blocking RFC 1918 prefixes ###"},
+                    {"sequence": self._get_sequence(), "action": "deny", "protocol": "ip", "source": "any", "destination": "10.0.0.0/8"},
+                    {"sequence": self._get_sequence(), "action": "deny", "protocol": "ip", "source": "any", "destination": "172.16.0.0/12"},
+                    {"sequence": self._get_sequence(), "action": "deny", "protocol": "ip", "source": "any", "destination": "192.168.0.0/16"},
+                    {"sequence": self._get_sequence(), "action": "permit", "protocol": "ip", "source": "any", "destination": "any"},
+                ],
+            }
+            return outbound_access_list
+        return None
+
+    def _get_access_lists_inbound(self) -> List | None:
+        """
+        Returns the inbound access lists for different interfaces.
+        The ACL name per interface is of the form "WAN_TRANSPORT_<INTERFACE_NAME>_IN".
+        """
+        inbound_access_lists = []
+        for l3_interface in self.shared_utils.wan_interfaces:
+            interface_ip = l3_interface.get("ip_address")
+            # Need to see how to handle dhcp cases.
+            if interface_ip and interface_ip != "dhcp":
+                subnet = str(ip_network(interface_ip, False))
+                l3_interface_name = l3_interface.get("name")
+                inbound_access_list = {
+                    "name": self._get_inbound_wan_acl_name(l3_interface_name),
+                    "entries": [],
+                }
+                # Allow BGP
```

Comment:
BGP is only needed here for peering with service provider. This is not supported yet in AVD, so I don't think we should permit it in the ACL by default. (Later we can add it conditionally.